test/runtests.jl

```diff
@@ -140,3 +140,21 @@ end
     @test (identity.(Not(1)) .=> [sin, cos] .=> Not(1)) ==
           [InvertedIndices.BroadcastedInvertedIndex(Not(1)) => sin => InvertedIndices.BroadcastedInvertedIndex(Not(1)), InvertedIndices.BroadcastedInvertedIndex(Not(1)) => cos => InvertedIndices.BroadcastedInvertedIndex(Not(1))]
 end
+
+@testset "tuples" begin
+    if VERSION >= v"1.7-dev"
+        nt = (a=1, b="2", c=:c)
+        @test nt[Not(:a)] === nt[(:b, :c)]
+        @test nt[Not((:a,))] === nt[(:b, :c)]
+        @test nt[Not((:a, :b))] === nt[(:c,)]
+        @test nt[Not(())] === nt
+        @test nt[Not(:xxx)] === nt
```

Comment:
For vectors we have the following behavior:
```
julia> x = 1:10
1:10

julia> x[Not(11)]
ERROR: BoundsError: attempt to access 10-element UnitRange{Int64} at index [Not(11)]

julia> x[Not([1,11])]
ERROR: BoundsError: attempt to access 10-element UnitRange{Int64} at index [Not([1, 11])]
```

Reply:
Currently, I followed a kind of prior art for namedtuples:
```
julia> nt = (a=1, b=2)

julia> Base.structdiff(nt, NamedTuple{(:a, :xxx)})
(b = 2,)

julia> using Accessors
julia> @delete nt.xxx
(a = 1, b = 2)
```
Basically, integer-indexed and name-indexed collections tend to behave differently wrt nonexistent indices. Not only (named)tuples, but also vectors vs dicts:
```
julia> deleteat!([1, 2, 3], 4)
ERROR: BoundsError: attempt to access 3-element Vector{Int64} at index [4]

julia> delete!(Dict(1 => 1, 2 => 2, 3 => 3), 4)
Dict{Int64, Int64} with 3 entries:
  2 => 2
  3 => 3
  1 => 1
```
So I thought it makes most sense to follow this convention with `Not()` as well.